src/blocks/tooltip/Tooltip.types.ts

```diff
@@ -0,0 +1,69 @@
+import type { ReactNode } from 'react';
+
+import type { BlockWithoutStyleProp, ResponsiveProp, SpaceType, ValueOf } from '../Blocks.types';
+import type { FlattenSimpleInterpolation } from 'styled-components';
+
+export type TooltipResponsiveProps = {
+  /* Sets height css property */
+  height?: ResponsiveProp<string>;
+  /* Sets margin css property */
+  margin?: ResponsiveProp<SpaceType>;
+  /* Sets max-height css property */
+  maxHeight?: ResponsiveProp<string>;
+  /* Sets min-height css property */
+  minHeight?: ResponsiveProp<string>;
+  /* Sets max-width css property */
+  maxWidth?: ResponsiveProp<string>;
+  /* Sets min-width css property */
+  minWidth?: ResponsiveProp<string>;
+  /* Sets width css property */
+  width?: ResponsiveProp<string>;
+};
+
+export type TooltipPosition = 'top-right' | 'top-left' | 'bottom-right' | 'bottom-left';
+
+export type TooltipComponentProps = {
+  /* Additional prop from styled components to apply custom css to Tooltip */
+  css?: FlattenSimpleInterpolation;
+  /* Child react nodes rendered by Tooltip */
+  children?: ReactNode;
+  /* Overlay content component of the Tooltip */
+  overlay?: ReactNode;
+  /* Title of the Tooltip */
+  title?: string;
+  /* Description to be displayed in the tooltip */
+  description?: string;
+  /* Position of the Tooltip */
+  tooltipPosition?: TooltipPosition;
+  /* Boolean value indicating whether Tooltip is visible */
+  visible?: boolean;
+  /* Width of the tooltip container */
+  containerWidth?: number;
+  /* Height of the toolip container */
+  containerHeight?: number;
+};
+
+export type TooltipContainerProps = {
+  /* Trigger for the Tooltip to open */
+  trigger?: 'hover' | 'click';
```

Comment:
This should accept an array as well apart from string. So we can pass both hover and click as trigger.
In mobile hover doesn't works and only click works. So it will be helpful in that case.

Reply:
I think you can use this library to develop tooltip since there are use cases which we might not able to cover without a library. https://reach.tech/tooltip

Please see how the behaviour for click and hover is developed in it.